WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -224,6 +224,73 @@ final class ShippingLabelFormViewModel {
 
         return String.localizedStringWithFormat(Localization.paymentMethodLabel, selectedPaymentMethod.cardDigits)
     }
+
+    /// Returns the subtotal under the Order Summary.
+    ///
+    func getSubtotal() -> String {
+        guard let selectedRate = selectedRate else {
+            return ""
+        }
+
+        var retailRate: Double = selectedRate.retailRate
+        if let selectedSignatureRate = selectedSignatureRate {
+            retailRate = selectedSignatureRate.retailRate
+        }
+        else if let selectedAdultSignatureRate = selectedAdultSignatureRate {
+            retailRate = selectedAdultSignatureRate.retailRate
+        }
+
+        let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
+        let price = currencyFormatter.formatAmount(Decimal(retailRate)) ?? ""
+
+        return price
+    }
+
+    /// Returns, if available, the discount under the Order Summary.
+    ///
+    func getDiscount() -> String? {
+        guard let selectedRate = selectedRate else {
+            return nil
+        }
+
+        var rate: Double = selectedRate.retailRate - selectedRate.rate
```

Comment:
I noticed that the discount appears as a positive number if I select an expensive shipping rate without adding any signature option. It looks like `retailRate` and `rate` should be switched here:

```suggestion
        var rate: Double = selectedRate.rate - selectedRate.retailRate
```

Reply:
Good catch!